R/textmodel_lr.R

```diff
@@ -0,0 +1,215 @@
+#' Logistic regression classifier for texts
+#'
+#' Fit a fast logistic regression classifier for texts, using the \pkg{glmnet}
+#' package.
+#' @param x the \link{dfm} on which the model will be fit.  Does not need to
+#'   contain only the training documents.
+#' @param y vector of training labels associated with each document identified
+#'   in \code{train}.  (These will be converted to factors if not already
+#'   factors.)
+#' @param nfolds .
+#' @param ... additional arguments passed to \code{\link[glmnet]{cv.glmnet}}
+#' @seealso \code{\link[glmnet]{cv.glmnet}}
+#' @examples
+#' \dontrun{
+#' library(quanteda.textmodels)
+#' # use party leaders for govt and opposition classes
+#' quanteda::docvars(data_corpus_irishbudget2010, "govtopp") <-
+#'     c(rep(NA, 4), "Govt", "Opp", NA, "Opp", NA, NA, NA, NA, NA, NA)
+#' dfmat <- quanteda::dfm(data_corpus_irishbudget2010)
+#' tmod <- textmodel_lr(dfmat, y = docvars(dfmat, "govtopp"))
```

Comment:
This didn't work for me

Reply:
Sorry about that. I included the wrong example. Instead, I used the one from `textmodel_nb()` now and adapted it slightly. The problem with the original example is that lr needs at least two cases per class (the original example has just one for negative) and roughly 50 cases in total (so I used `dfm_sample()`; also see the tests). The underlying `cv.glmnet()` has some errors about this baked in but I might need to add a clearer warning.